pkg/sql/resolver.go

```diff
@@ -265,12 +265,15 @@ func (p *planner) HasAnyPrivilege(
 		}
 
 		if priv.GrantOption {
-			if err := p.CheckGrantOptionsForUser(ctx, desc.GetPrivileges(), desc, []privilege.Kind{priv.Kind}, user, true /* isGrant */); err != nil {
-				if pgerror.GetPGCode(err) == pgcode.WarningPrivilegeNotGranted {
-					continue
-				}
+			isGrantable, err := p.CheckGrantOptionsForUser(
+				ctx, desc.GetPrivileges(), desc, []privilege.Kind{priv.Kind}, user,
+			)
+			if err != nil {
 				return eval.HasNoPrivilege, err
```

Comment:
Moot point but maybe `eval.ObjectNotFound` is more appropriate? `CheckGrantOptionsForUser` encountered a runtime error, so you don't exactly know if that implies `HasNoPrivilege`, right? In any case, the caller is not allowed to make any assumption about the first result arg. since `err != nil`.

Reply:
Realistically, I think it'd be even better if `0` was not `eval.HasNoPrivilege` but rather was either not defined as a value. I don't think `eval.ObjectNotFound` is better because it's non-zero, and it in practice it implies that that the object doesn't eixst whereas `HasNoPrivilege` doesn't definitely imply that it does. Generally the idiom is that when returning an error you should return a zero value for the other fields if they are not intended for use. This code could all be better in terms of its types.